### PR TITLE
Nltk Removed

### PR DESCRIPTION
**Pull request by erice22**

### CODE REVIEW

File: mentor_classifier/requirements.txt [hunk 1]
```diff
@@ -10,5 +10,8 @@ PyYAML==5.4.1
 redis==3.5.3
 requests==2.25.1
 scikit-learn==0.24.2
-sentence-transformers==2.0.0
+sentence-transformers
```
Comment (beatthat) [inline mentor_classifier/requirements.txt:13]: do we want sentence-transformers here or next PR?

File: mentor_classifier/mentor_classifier/__init__.py [hunk 2]
```diff
@@ -40,14 +40,14 @@ class QuestionClassiferPredictionResult:
 
 class QuestionClassifierTraining(ABC):
     @abstractmethod
-    def train(self) -> QuestionClassifierTrainingResult:
+    def train(self, shared_root) -> QuestionClassifierTrainingResult:
```
Comment (beatthat) [inline mentor_classifier/mentor_classifier/__init__.py:43]: `shared_root: str`

File: mentor_classifier/mentor_classifier/arch/lr/predict.py [hunk 3]
```diff
@@ -44,7 +44,10 @@ def __init__(self, mentor, shared_root, data_path):
         self.model = self.__load_model()
 
     def evaluate(
-        self, question, canned_question_match_disabled=False
+        self,
+        question,
```
Comment (beatthat) [inline mentor_classifier/mentor_classifier/arch/lr/predict.py:48]: set types on params

File: mentor_classifier/mentor_classifier/arch/lr_transformer/train.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def __init__(
         self.model_path = mentor_model_path(output_dir, mentor.id, ARCH_LR_TRANSFORMER)
         self.transformer = TransformerEmbeddings(shared_root)
 
-    def train(self) -> QuestionClassifierTrainingResult:
+    def train(self, shared_root) -> QuestionClassifierTrainingResult:
```
Comment (beatthat) [inline mentor_classifier/mentor_classifier/arch/lr_transformer/train.py:48]: set types

File: mentor_classifier/mentor_classifier/spacy_model.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+#
+# The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+#
+from os import path
+from spacy import load
+
+SPACY_MODELS = {}
+
+
+def find_or_load_spacy(file_path: str):
```
Comment (beatthat) [inline mentor_classifier/mentor_classifier/spacy_model.py:13]: return type (must be some spacy class)

File: mentor_classifier/tests/fixtures/data/ner/test.csv [hunk 6]
```diff
@@ -0,0 +1,2 @@
+Sentence, entities
+"My name is EMC Clint Anderson, that's Electrician's Mate Clinton Anderson. I was born in Los Angeles, California. I was raised there most of my life and I graduated from high school there. A couple of years after graduating from high school, then I joined the United States Navy. I was an Electrician's Mate for eight years. I served on an aircraft carrier. We went on many deployments. A deployment is when you go to war, you fight. We fought in the Iraq war. I went on three deployments and it was a really great time in my life. I had a lot of fun. At the end of the eight years, I decided that the Navy wasn't quite a career for me. So, I got out of the Navy. I started using the education benefits that we received and I started going to the University of California at Berkeley. I was majoring in computer science and afterwards, I started getting my master's degree from the University of Southern California. I also had a job at the Institute for Creative Technologies. It's been a lot of fun, this whole time. Thanks to the Navy."
```
Comment (beatthat) [inline mentor_classifier/tests/fixtures/data/ner/test.csv:2]: new line at EOF

File: mentor_classifier/tests/helpers.py [hunk 7]
```diff
@@ -106,6 +112,9 @@ def run_model_against_testset(
         if test_set_entry.comparison_type == ComparisonType.GTE:
             if test_result.highest_confidence < test_set_entry.expected_confidence:
                 current_result_entry.passing = False
+                import logging
```
Comment (beatthat) [inline mentor_classifier/tests/helpers.py:115]: let's remove these logs

File: mentor_classifier_api/Dockerfile [hunk 8]
```diff
@@ -20,9 +20,5 @@ RUN rm -rf /tmp/build
 ENV FLASK_APP=mentor_classifier_api
 WORKDIR /app
 COPY src .
-RUN python -m nltk.downloader punkt \
```
Comment (beatthat) [inline mentor_classifier_api/Dockerfile:23]: is not a new problem, but add NEW LINE at EOF

**Review by beatthat — CHANGES_REQUESTED** [review]
This looks great! #1 issue is let's remove `sentence-transformers` if we don't need it yet, and then just add all the missing types

File: mentor_classifier/requirements.txt (hunk 1 above)
Comment (erice22) [reply mentor_classifier/requirements.txt:13]: we still need sentence-transformers for the lr_transformer arch

Comment (beatthat) [reply mentor_classifier/requirements.txt:13]: ok, in that case, just make sure we specify the version ,`sentence-transformers==2.0.0`...and we then need to include the changes to both Dockerfiles that do the cpu-only pip installs for py torch. Are you able to find those changes and add to this branch?

**Review by beatthat — APPROVED** [review]
looks great! excited for this

**Pull request merged**